src/replica.rs

```diff
@@ -399,10 +417,9 @@ impl Replica {
     /// system
     pub fn sync(&mut self, server: &mut Box<dyn Server>, avoid_snapshots: bool) -> Result<()> {
         self.taskdb
-            .sync(server, avoid_snapshots)
+            .sync(self.storage.txn()?.as_mut(), server, avoid_snapshots)
             .context("Failed to synchronize with server")?;
-        self.rebuild_working_set(false)
-            .context("Failed to rebuild working set after sync")?;
+        self.rebuild_working_set(false)?;
```

Comment:
This seems to have lost the context?

```suggestion
        self.rebuild_working_set(false)
            .context("Failed to rebuild working set after sync")?;
```